### DISCUSSION TITLE
--import-config does not work

### DISCUSSION THREAD

**Discussion by danielmarschall** [rustdesk/rustdesk#6903] [Q&A]
Hello,

I have an RustDesk2.toml file which I want to import.

I execute `rustdesk.exe --import-config RustDesk.toml` .

Using Process Monitor, I could verify that the config file is read by the RustDesk.exe process:

![image](https://github.com/rustdesk/rustdesk/assets/28412477/40e630d4-05f2-4b31-9f93-4df8c0636469)

The log file `c:\Users\test\AppData\Roaming\RustDesk\log\import-config\` says:

    [2024-01-16 16:38:26.824175 +01:00] INFO [src\core_main.rs:162] main start args:["--import-config", "RustDesk.toml"]
    [2024-01-16 16:38:26.828150 +01:00] INFO [src\core_main.rs:457] import config from "C:\\CORA2012\\BAP\\bin\\RustDesk.toml" and "C:\\CORA2012\\BAP\\bin\\RustDesk2.toml"

According to the [source code](https://github.com/rustdesk/rustdesk/blob/master/src/core_main.rs#L504), there should be a log message "config2 written" if
1. `get_modified_time(&path2) > get_modified_time(&Config2::file())`
2. `store_path(Config2::file(), config2).is_err()`

I do not know which of these two conditions fail.

I think that the first condition should be fine. I even faked the modification time to "2 days in the future", just to be 100% sure that `get_modified_time` is always bigger to whatever `&Config2::file()` is

What can I do?

Thank you for the help.

---

Background information:  Why do I want to use `--import-config` ?

Various users execute `rustdesk.exe` with different Windows user profiles.  ("Run as other user" or "Run as administrator")
I do not want to change `c:\Users\...\AppData\Roaming\RustDesk\config\` for every user on the system.
So my idea is to put a `Rustdesk2.toml` file somewhere world-readable on the harddisk and the users execute `rustdesk.exe --import-config ...` to start rustdesk using their user profile but with an individual server/key.

**Comment by kamaranl** [comment]
I was needing this to work as well, but for different reasons, and found out during my trials that ***both*** _Rustdesk.toml_ and _Rustdesk2.toml_ must be present in the directory that you are importing Rustdesk.toml from.

My process went something like this:

1. Downloaded RustDesk to `$HOME\Downloads`.
2. Launch RustDesk without any parameters.
3. Navigate GUI to configure ID server.
4. Close RustDesk.
5. Navigate to `$env:APPDATA\RustDesk\config\` and copy both RustDesk.toml and RustDesk2.toml to another directory- say `$HOME\Downloads`.
6. Delete `$env:APPDATA\RustDesk`.
7. Open up a terminal tab at `$HOME\Downloads` and run:
```powershell
./rustdesk.exe --import-config RustDesk.toml
sleep 1
./rustdesk.exe
```

RustDesk then had my configured ID server, as expected.

Hope that helps.
